File: isuite-core-ui/src/main/java/gov/nwcg/isuite/core/service/ReportCostService.java
package gov.nwcg.isuite.core.service;

import gov.nwcg.isuite.core.reports.filter.AircraftDetailReportFilter;
import gov.nwcg.isuite.core.reports.filter.AnalysisReportFilter;
import gov.nwcg.isuite.core.reports.filter.CostShareReportFilter;
import gov.nwcg.isuite.core.reports.filter.GroupCategoryTotalReportFilter;
import gov.nwcg.isuite.core.vo.dialogue.DialogueVo;
import gov.nwcg.isuite.framework.exceptions.PersistenceException;
import gov.nwcg.isuite.framework.exceptions.ServiceException;


public interface ReportCostService {
	
	/**
	 * 
	 * @param filter
	 * @param incidentId
	 * @param incidentResourceId
	 * @return
	 * @throws ServiceException
	 * @throws PersistenceException 
	 */
	public DialogueVo generateAircraftDetailReport(AircraftDetailReportFilter filter,DialogueVo dialogueVo) throws Exception;

	public DialogueVo generateGroupCategorySummaryReport(GroupCategoryTotalReportFilter filter,DialogueVo dialogueVo) throws Exception;

	public DialogueVo generateSummaryByResourceReport(GroupCategoryTotalReportFilter filter, DialogueVo dialogueVo) throws Exception;

	public DialogueVo generateSummaryForCurrentDayReport(GroupCategoryTotalReportFilter filter, DialogueVo dialogueVo) throws Exception;

	public DialogueVo generateDetailByResourceReport(GroupCategoryTotalReportFilter filter, DialogueVo dialogueVo) throws ServiceException ;

	public DialogueVo generateAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateDailyCostComparisonICAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateDailyCostComparisonRESRAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateResourcesWithDailyCostExceeds10000AnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateResourcesWithActualTimePostingButThreeOrMoreDaysAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateResourcesWithNoActualTimePostedAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateResourcesWithNoDailyCostRecordsAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateResourceItemCodeByCostAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generateResourceItemCodeByCostOHPersonnelAnalysisReport(AnalysisReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo getGroupCategoryTotalFilter(Long id, String filterName, boolean isIncidentGroup, DialogueVo dialogueVo) throws Exception;
	
	public DialogueVo generateGroupCategoryTotalReport(GroupCategoryTotalReportFilter filter,DialogueVo dialogueVo) throws Exception;
	
	public DialogueVo generateCostShareReport(CostShareReportFilter filter,DialogueVo dialogueVo) throws Exception;
}

File: eis2/src/app/service/report-cost.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { DialogueVo } from 'src/app/data/dialogue/dialoguevo';
import { HttpParams, HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class ReportCostService {
  constructor(private http: HttpClient) { }

  generateGroupCategorySummaryReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/group-category-summary', body);
  }

  generateSummaryByResourceReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/summary-by-resource', body);
  }

  generateSummaryForCurrentDayReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/summary-for-current-day', body);
  }

  generateDetailByResourceReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/details-by-resource', body);
  }

  generateGroupCategoryTotalReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/group-category-total-by-filter', body);
  }

  generateAircraftDetailReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/aircraft-detail', body);
  }

  generateAnalysisReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/analysis', body);
  }

  generateCostShareReport(body): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/costs/cost-share', body);
  }

  getGroupCategoryTotalFilter(id, filternName, isIncidentGroup): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
      .set('id', id)
      .set('filternName', filternName)
      .set('isIncidentGroup', isIncidentGroup);
    return this.http.post<DialogueVo>('/service/reports/costs/group-category-total', params);
  }
}
